File: dashbord/src/services/appealService.js
import axios from 'src/lib/axios';
import { buildApiPath } from 'src/services/backend';




export const appealService = {
  // Get all appeals (admin)
  getAllAppeals: async (params = {}) => {
    try {
      const response = await axios.get(buildApiPath('/appeals/admin'), { params });
      return response.data;
    } catch (error) {
      console.error('Error getting appeals:', error);
      throw error;
    }
  },

  // Get appeal details
  getAppealDetails: async (id) => {
    try {
      const response = await axios.get(buildApiPath(`/appeals/admin/${id}`));
      return response.data;
    } catch (error) {
      console.error('Error getting appeal details:', error);
      throw error;
    }
  },

  // Update appeal status
  updateAppealStatus: async (id, status, adminResponse) => {
    try {
      const response = await axios.patch(buildApiPath(`/appeals/admin/${id}`), {
        status,
        admin_response: adminResponse,
      });
      return response.data;
    } catch (error) {
      console.error('Error updating appeal status:', error);
      throw error;
    }
  },

  // Get appeal statistics
  getAppealStats: async () => {
    try {
      const response = await axios.get(buildApiPath('/appeals/admin/stats'));
      return response.data;
    } catch (error) {
      console.error('Error getting appeal stats:', error);
      throw error;
    }
  },

  // Delete appeal
  deleteAppeal: async (id) => {
    try {
      const response = await axios.delete(buildApiPath(`/appeals/admin/${id}`));
      return response.data;
    } catch (error) {
      console.error('Error deleting appeal:', error);
      throw error;
    }
  },

  // Get user appeals
  getUserAppeals: async (params = {}) => {
    try {
      const response = await axios.get(buildApiPath('/appeals/me'), { params });
      return response.data;
    } catch (error) {
      console.error('Error getting user appeals:', error);
      throw error;
    }
  },

  // Submit appeal
  submitAppeal: async (appealData) => {
    try {
      const response = await axios.post(buildApiPath('/appeals'), appealData);
      return response.data;
    } catch (error) {
      console.error('Error submitting appeal:', error);
      throw error;
    }
  },

  // Submit appeal without authentication (suspended/banned users)
  submitAnonymousAppeal: async (appealData) => {
    try {
      const response = await axios.post(buildApiPath('/appeals/anonymous'), appealData);
      return response.data;
    } catch (error) {
      console.error('Error submitting anonymous appeal:', error);
      throw error;
    }
  },
};

File: Front/lib/services/appeal_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import 'api_client.dart';

class AppealService {
  static Future<Map<String, dynamic>> submitAppeal({
    required String subject,
    required String message,
    List<String>? attachments,
  }) async {
    try {
      final response = await ApiClient.post('/appeals', {
        'subject': subject,
        'message': message,
        'attachments': attachments ?? [],
      });

      Map<String, dynamic> body = {};
      try {
        body = jsonDecode(response.body);
      } catch (_) {
        body = {};
      }

      return {
        'success': response.statusCode == 201,
        'message': body['message'] ?? 'Unable to submit appeal',
        'appeal': body['appeal'],
      };
    } catch (_) {
      return {'success': false, 'message': 'Unable to submit appeal right now.'};
    }
  }

  static Future<Map<String, dynamic>> getUserAppeals({
    String? status,
    int limit = 10,
  }) async {
    try {
      String queryParams = '';
      if (status != null) {
        queryParams += '?status=$status';
      }
      if (limit != 10) {
        queryParams += queryParams.isEmpty ? '?limit=$limit' : '&limit=$limit';
      }

      final response = await ApiClient.get('/appeals/me$queryParams');
      
      Map<String, dynamic> body = {};
      try {
        body = jsonDecode(response.body);
      } catch (_) {
        body = {};
      }

      return {
        'success': response.statusCode == 200,
        'appeals': body['appeals'] ?? [],
        'count': body['count'] ?? 0,
      };
    } catch (_) {
      return {'success': false, 'appeals': [], 'count': 0};
    }
  }

  static Future<Map<String, dynamic>> getAppealDetails(String appealId) async {
    try {
      final response = await ApiClient.get('/appeals/admin/$appealId');
      
      Map<String, dynamic> body = {};
      try {
        body = jsonDecode(response.body);
      } catch (_) {
        body = {};
      }

      return {
        'success': response.statusCode == 200,
        'appeal': body['appeal'],
      };
    } catch (_) {
      return {'success': false, 'appeal': null};
    }
  }

  static Future<Map<String, dynamic>> getAllAppeals({
    String? status,
    String? search,
    int limit = 20,
    int skip = 0,
  }) async {
    try {
      List<String> queryParams = [];
      
      if (status != null) queryParams.add('status=$status');
      if (search != null) queryParams.add('search=$search');
      if (limit != 20) queryParams.add('limit=$limit');
      if (skip != 0) queryParams.add('skip=$skip');

      final queryString = queryParams.isNotEmpty ? '?${queryParams.join('&')}' : '';
      
      final response = await ApiClient.get('/appeals/admin$queryString');
      
      Map<String, dynamic> body = {};
      try {
        body = jsonDecode(response.body);
      } catch (_) {
        body = {};
      }

      return {
        'success': response.statusCode == 200,
        'appeals': body['appeals'] ?? [],
        'total': body['total'] ?? 0,
        'limit': body['limit'] ?? limit,
        'skip': body['skip'] ?? skip,
        'hasMore': body['hasMore'] ?? false,
      };
    } catch (_) {
      return {
        'success': false,
        'appeals': [],
        'total': 0,
        'limit': limit,
        'skip': skip,
        'hasMore': false,
      };
    }
  }

  static Future<Map<String, dynamic>> updateAppealStatus(
    String appealId,
    String status, {
    String? adminResponse,
  }) async {
    try {
      final response = await ApiClient.patch('/appeals/admin/$appealId', {
        'status': status,
        'admin_response': adminResponse,
      });

      Map<String, dynamic> body = {};
      try {
        body = jsonDecode(response.body);
      } catch (_) {
        body = {};
      }

      return {
        'success': response.statusCode == 200,
        'message': body['message'] ?? 'Unable to update appeal status',
        'appeal': body['appeal'],
      };
    } catch (_) {
      return {'success': false, 'message': 'Unable to update appeal status right now.'};
    }
  }

  static Future<Map<String, dynamic>> getAppealStats() async {
    try {
      final response = await ApiClient.get('/appeals/admin/stats');
      
      Map<String, dynamic> body = {};
      try {
        body = jsonDecode(response.body);
      } catch (_) {
        body = {};
      }

      return {
        'success': response.statusCode == 200,
        'pending': body['pending'] ?? 0,
        'reviewed': body['reviewed'] ?? 0,
        'accepted': body['accepted'] ?? 0,
        'rejected': body['rejected'] ?? 0,
        'last24h': body['last24h'] ?? 0,
      };
    } catch (_) {
      return {
        'success': false,
        'pending': 0,
        'reviewed': 0,
        'accepted': 0,
        'rejected': 0,
        'last24h': 0,
      };
    }
  }
}
